### ISSUE TITLE
Issue with GradientExplainer and TF2

### ISSUE THREAD

**Issue by antoncic** [shap/shap#986]
Hi, 
I am trying to get the TF1 example of Gradient Explainer in the readme to work in TF2 but I fail miserably...
Have some one got that to work? 

My python setup looks like so:
```
conda create -n tf2 python=3.7
conda activate tf2
conda install nomkl
conda install matplotlib pillow tensorflow
conda install -c conda-forge shap

MacOS version = Catalina 10.15.2
TensorFlow version = 2.0.0
shap version = 0.34
```

My attempt looks like so:
 ```
import tensorflow as tf
import numpy as np
import json
import shap

# load pre-trained model and choose two images to explain
model = tf.keras.applications.VGG16(weights='imagenet', include_top=True)

# load the ImageNet background images
![TF1-shap-layer-0](https://user-images.githubusercontent.com/3359645/71853137-d010ff00-30da-11ea-9108-1dc43f2fcaba.png)

X,y = shap.datasets.imagenet50()
to_explain = X[[40,41]]
    
# Explain
e = shap.GradientExplainer(model, X)
shap_values,indexes = e.shap_values(to_explain, ranked_outputs=2)

# load the ImageNet class names
url = "https://s3.amazonaws.com/deep-learning-models/image-models/imagenet_class_index.json"
fname = shap.datasets.cache(url)
with open(fname) as f:
    class_names = json.load(f)

# get the names for the classes
index_names = np.vectorize(lambda x: class_names[str(x)][1])(indexes)

# plot the explanations
shap.image_plot(shap_values, to_explain, index_names)
```
which does not throw errors. But it has two problems,

1. It does not generate the same result as in the TF1 case (when setting the layer to explain to 0 in TF1 example) and is probably wrong. See my attached results.
2. I cannot, after two days of banging my head against the wall, understand how I should explain intermediate layers as in the TF1 example. 

Why am I not just using TF1? Because I have built a model using transfer learning and a pretrained network in TF2 that I want to explain and I cannot get that to work in the TF1 case when I load it from the h5 file.

Help incredibly aprieciated!

TF1 Result for layer 0
![TF1-shap-layer-0](https://user-images.githubusercontent.com/3359645/71853208-f9318f80-30da-11ea-8271-674d1b9ddfa4.png)
TF2 Result for the code above
![TF2-shap-layer-0](https://user-images.githubusercontent.com/3359645/71853209-f9318f80-30da-11ea-8a30-a9291ac20045.png)

**Comment by riemann85** [comment]
Hi, thank you for your explainability package.
Are you aware of any libraries for interpretability working with Tensorflow 2.0 currently?
I tried shap , deepexplain tf-explain but many have compatibility problems with Tf2.
Thanks
